pkg/scheduler/plugins/coescheduling/core/core.go

```diff
@@ -0,0 +1,618 @@
+				waitingPod.Reject(pluginName, "gang rejection in PostFilter")
+			}
+		})
+		gang.setScheduleCycleValid(false)
+		// since we rejected the pod which is waiting in Permit stage, we need to clear the gang's TimeoutStartTime field
+		gang.clearTimeoutStartTime()
+		return &framework.PostFilterResult{}, framework.NewStatus(framework.Unschedulable,
+			fmt.Sprintf("Gang: %v gets rejected this cycle due to Pod: %v is unschedulable even after "+
+				"PostFilter in StrictMode", gangId, pod.Name))
+	}
+
+	return &framework.PostFilterResult{}, framework.NewStatus(framework.Unschedulable, "")
+}
+
+// PostBind updates a PodGroup's status.
+func (pgMgr *PodGroupManager) PostBind(ctx context.Context, pod *corev1.Pod, nodeName string) {
+	// first update CRD
+	if !PgFromAnnotation(pod) {
+		pgFullName, pg := pgMgr.GetPodGroup(pod)
+		if pgFullName == "" || pg == nil {
+			return
+		}
+		pgCopy := pg.DeepCopy()
+		pgCopy.Status.Scheduled++
+
+		if pgCopy.Status.Scheduled >= pgCopy.Spec.MinMember {
+			pgCopy.Status.Phase = v1alpha1.PodGroupScheduled
+		} else {
+			pgCopy.Status.Phase = v1alpha1.PodGroupScheduling
+			if pgCopy.Status.ScheduleStartTime.IsZero() {
+				pgCopy.Status.ScheduleStartTime = metav1.Time{Time: time.Now()}
+			}
+		}
+		if pgCopy.Status.Phase != pg.Status.Phase {
+			pg, err := pgMgr.pgLister.PodGroups(pgCopy.Namespace).Get(pgCopy.Name)
+			if err != nil {
+				klog.Errorf("Failed to get PodGroup: %v, err: %v",
+					GetNamespaceSplicingName(pgCopy.Namespace, pgCopy.Name), err)
+				return
+			}
+			patch, err := CreateMergePatch(pg, pgCopy)
+			if err != nil {
+				klog.Errorf("Failed to create merge patch,podGroup: %v, err: %v",
+					GetNamespaceSplicingName(pgCopy.Namespace, pgCopy.Name), err)
+				return
+			}
+			if err := pgMgr.PatchPodGroup(pg.Name, pg.Namespace, patch); err != nil {
+				klog.Errorf("Failed to patch podGroup: %v, err: %v",
+					GetNamespaceSplicingName(pgCopy.Namespace, pgCopy.Name), err)
+				return
+			}
+		}
+	}
+	// update gang in cache
+	gangName := GetGangNameByPod(pod)
+	if gangName == "" {
+		return
+	}
+	gangId := GetNamespaceSplicingName(pod.Namespace,
+		gangName)
+	gang := pgMgr.cache.getGangFromCache(gangId)
+	if gang == nil {
+		klog.Errorf("can't find gang, gangName: %v, podName: %v", gangId,
+			GetNamespaceSplicingName(pod.Namespace, pod.Name))
+		return
+	}
+	gang.addBoundPod(pod)
+}
+
+func (pgMgr *PodGroupManager) Unreserve(ctx context.Context, state *framework.CycleState, pod *corev1.Pod, nodeName string, handle framework.Handle, pluginName string) {
+	gangName := GetGangNameByPod(pod)
+	if gangName == "" {
+		return
+	}
+	gangId := GetNamespaceSplicingName(pod.Namespace,
+		gangName)
+	gang := pgMgr.cache.getGangFromCache(gangId)
+	if gang == nil {
+		klog.Errorf("can't find gang, gangName: %v, podName: %v", gangId,
+			GetNamespaceSplicingName(pod.Namespace, pod.Name))
+		return
+	}
+
+	// gang time out
+	if gang.isGangTimeout() {
```

Comment:
is resource is satisfied, you shouldn't handle Unreserve, this may happen when total > min, and min saitisfied, then pod comes to Unreserve stage.

Reply:
ok